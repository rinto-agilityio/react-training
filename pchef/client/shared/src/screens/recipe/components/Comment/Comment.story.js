// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import Comment from '.'

// Helper
import { wInfo } from '../../../../../.storybook/utils';

storiesOf('Comment', module)
  .addDecorator(wInfo())
  .add('Primary', () => (
    <Comment
      name="Andy"
      publishedAt={Date.now()}
      text="Yummy"
    />
  ))
  .add('Secondary', () => (
    <Comment
      type="secondary"
      name="Andy"
      publishedAt={Date.now()}
      text="Yummy"
    />
  ))
