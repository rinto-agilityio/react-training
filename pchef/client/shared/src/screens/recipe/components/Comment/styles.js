// Libs
import { StyleSheet } from 'react-native'

// Themes
import { METRICS, FONTS } from '../../../../themes'

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingLeft: METRICS.mediumPadding,
    paddingRight: METRICS.mediumPadding,
    paddingTop: METRICS.mediumPadding,
    paddingBottom: METRICS.mediumPadding,
  },
  dash: {
    marginLeft: METRICS.smallMargin,
    marginRight: METRICS.smallMargin,
  },
  userAvatar: {
    marginRight: METRICS.mediumMargin,
  },
  commentWrapper: {
    flex: 1,
  },
  userInfo: {
    flexDirection: 'row',
  },
  primaryInfor: {
    fontSize: FONTS.fontSize.small,
  },
  secondaryInfor: {
    fontSize: FONTS.fontSize.base,
  },
  primaryContent: {
    fontSize: FONTS.fontSize.base,
    marginTop: METRICS.smallMargin,
  },
  secondaryContent: {
    fontSize: FONTS.fontSize.medium,
    marginTop: METRICS.mediumMargin,
  },
})
