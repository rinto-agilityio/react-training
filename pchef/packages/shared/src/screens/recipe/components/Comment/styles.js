import { StyleSheet } from 'react-native'
import { METRICS } from '../../../../themes'

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingLeft: METRICS.mediumPadding,
    paddingRight: METRICS.mediumPadding,
    paddingTop: METRICS.mediumPadding,
    paddingBottom: METRICS.mediumPadding,
  },
  dash: {
    marginLeft: METRICS.smallMargin,
    marginRight: METRICS.smallMargin,
  },
  userAvatar: {
    marginRight: METRICS.mediumMargin,
  },
  commentWrapper: {
    flex: 1,
  },
  userInfo: {
    flexDirection: 'row',
  },
  primaryInfor: {
    fontSize: METRICS.fontSize.small,
  },
  secondaryInfor: {
    fontSize: METRICS.fontSize.base,
  },
  primaryContent: {
    fontSize: METRICS.fontSize.base,
    marginTop: METRICS.smallMargin,
  },
  secondaryContent: {
    fontSize: METRICS.fontSize.medium,
    marginTop: METRICS.mediumMargin,
  },
});
