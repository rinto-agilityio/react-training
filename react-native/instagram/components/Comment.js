import React from "react";
import { View, TextInput, Image, StyleSheet } from "react-native";

export default class Comment extends React.Component {
  constructor(props) {
    super(props);
    this.state = { text: "" };
  }

  submitComment = () => {
    if (this.state.text) {
      const { postId, owner } = this.props;

      this.props.submitComment({
        postId: postId,
        userId: owner.id,
        text: this.state.text
      });

      this.setState({ text: "" });
    }
  };

  render() {
    const { owner } = this.props;

    return (
      <View style={styles.comment}>
        <Image style={styles.avatar} source={{ uri: owner.profile_pic_url }} />
        <TextInput
          style={styles.commentInput}
          placeholder="Add a comment"
          value={this.state.text}
          onChangeText={text => this.setState({ text })}
          onSubmitEditing={this.submitComment}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  comment: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    padding: 10
  },
  avatar: {
    height: 20,
    width: 20,
    borderRadius: 10,
    marginRight: 5
  },
  commentInput: {
    height: 30,
    paddingLeft: 10,
    borderRadius: 15,
    borderColor: "#a7a7a7",
    borderWidth: 1,
    width: "95%"
  }
});
