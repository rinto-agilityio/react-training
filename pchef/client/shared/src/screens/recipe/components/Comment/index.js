// Libs
import React from 'react'
import { View, Text } from 'react-native'
import { Avatar } from 'react-native-paper'

// Helpers
import { getPublishedTime } from '../../../../helpers/date-time'

// Styles
import styles from './styles'

// Themes
import { METRICS } from '../../../../themes'

type Props = {
  avatarUri?: string,
  name: string,
  publishedAt: number,
  text?: string,
  customStyle?: {},
  type: string,
  isGetTime?: boolean,
  customNameStyles?: {},
  customContentStyles?: {},
}

const Comment = ({
  avatarUri,
  name,
  text,
  customStyle,
  type = 'primary',
  publishedAt,
  isGetTime,
  customNameStyles,
  customContentStyles,
}: Props) => {
  // get comment published at time
  const time = getPublishedTime(publishedAt)

  // define container styles
  const containerStyle = [styles.container, customStyle]

  // define avatar size follow type
  const avatarSize = type === 'primary' ? METRICS.mediumAvatar : METRICS.largeAvatar

  return (
    <View style={containerStyle}>
      <View style={styles.userAvatar}>
        <Avatar.Image
          size={avatarSize}
          source={{
            uri: avatarUri,
          }}
        />
      </View>
      <View style={styles.commentWrapper}>
        <View style={styles.userInfo}>
          <Text style={[styles[`${type}Infor`], customNameStyles]}>
            {name}
          </Text>
          {!isGetTime && (
            <>
              <Text style={[styles.dash, styles[`${type}Infor`]]}>
                -
              </Text>
              <Text style={styles[`${type}Infor`]}>
                {time}
              </Text>
            </>
          )}
        </View>
        <Text style={[styles[`${type}Content`], customContentStyles]}>
          {isGetTime ? `on ${time}` : text}
        </Text>
      </View>
    </View>
  )
}

Comment.defaultProps = {
  avatarUri: '',
  customStyle: {},
  isGetTime: false,
  customNameStyles: {},
  customContentStyles: {},
  text: '',
}

export default Comment
