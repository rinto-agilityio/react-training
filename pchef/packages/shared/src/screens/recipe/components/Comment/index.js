import React from 'react';
import { View, Text } from 'react-native';
import { Avatar } from 'react-native-elements';
import { getPublishedTime } from '../../../../helpers/date-time'
import styles from './styles';

// Comment props type
type Props = {
  avatarUri?: string,
  name: string,
  publishedAt: string,
  text: string,
  customStyle?: {},
  type?: 'primary' | 'secondary'
}

// component Comment
const Comment = ({
  avatarUri,
  name,
  text,
  customStyle,
  type = 'primary',
  publishedAt,
}: Props) => {
  // get comment published at time
  const time = getPublishedTime(publishedAt);

  // define container styles
  const containerStyle = [styles.container, customStyle]

  // define avatar size follow type
  const avatarSize = type === 'primary' ? 'small' : 'medium'

  return (
    <View style={containerStyle}>
      <View style={styles.userAvatar}>
        <Avatar
          rounded
          size={avatarSize}
          source={{
            uri: avatarUri,
          }}
        />
      </View>
      <View style={styles.commentWrapper}>
        <View style={styles.userInfo}>
          <Text style={styles[`${type}Infor`]}>
            {name}
          </Text>
          <Text style={[styles.dash, styles[`${type}Infor`]]}>
            -
          </Text>
          <Text style={styles[`${type}Infor`]}>
            {time}
          </Text>
        </View>
        <Text style={styles[`${type}Content`]}>
          {text}
        </Text>
      </View>
    </View>
  )
}

// component default props value
Comment.defaultProps = {
  avatarUri: '',
  customStyle: {},
  type: 'primary',
};

export default Comment;
