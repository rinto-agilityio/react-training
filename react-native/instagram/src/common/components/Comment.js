// Libs
import React from 'react'
import { View, TextInput, Image } from 'react-native'
import PropTypes from 'prop-types'

// Styles
import { styles } from './styles/CommentStyles'
import CommonStyles from '@themes/common'

class Comment extends React.Component {
  state = { text: '' }

  // Reset state after unmount
  componentWillUnmount() {
    this._resetTextInput()
  }

  // Add comment for this photo
  handleSubmitComment = () => {
    const { text } = this.state

    if (text) {
      const { postId, owner, submitComment } = this.props

      submitComment({
        owner,
        postId,
        text
      })

      this._resetTextInput()
    }
  }

  /* eslint-disable react/no-set-state */
  handleTextChange = text => {
    this.setState({ text })
  }

  // Reset input value after submit or component unmount
  _resetTextInput = () => {
    this.setState({ text: '' })
  }

  /* eslint-enable react/no-set-state */

  render() {
    const { owner } = this.props,
      { text } = this.state

    return (
      <View style={[CommonStyles.layoutRow, styles.comment]}>
        <Image source={{ uri: owner.profile_pic_url }} style={styles.avatar} />
        <TextInput
          onChangeText={this.handleTextChange}
          onSubmitEditing={this.handleSubmitComment}
          placeholder="Add a comment"
          style={styles.commentInput}
          underlineColorAndroid="transparent"
          value={text}
        />
      </View>
    )
  }
}

Comment.propTypes = {
  owner: PropTypes.object.isRequired,
  postId: PropTypes.number.isRequired,
  submitComment: PropTypes.func.isRequired
}

export default Comment
