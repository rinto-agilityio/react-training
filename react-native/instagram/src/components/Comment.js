import React from 'react'
import { View, TextInput, Image } from 'react-native'
import PropTypes from 'prop-types'

import { styles } from './styles/Comment'

class Comment extends React.Component {
  constructor() {
    super()
    this.state = { text: '' }
  }

  submitComment = () => {
    if (this.state.text) {
      const { postId, owner, submitComment } = this.props

      submitComment({
        postId: postId,
        owner: owner,
        text: this.state.text
      })

      this.setState({ text: '' })
    }
  }

  render() {
    const { owner } = this.props

    return (
      <View style={styles.comment}>
        <Image style={styles.avatar} source={{ uri: owner.profile_pic_url }} />
        <TextInput
          style={styles.commentInput}
          placeholder="Add a comment"
          value={this.state.text}
          onChangeText={text => this.setState({ text })}
          onSubmitEditing={this.submitComment}
        />
      </View>
    )
  }
}

Comment.propTypes = {
  postId: PropTypes.number.isRequired,
  owner: PropTypes.object.isRequired,
  submitComment: PropTypes.func
}

export default Comment
