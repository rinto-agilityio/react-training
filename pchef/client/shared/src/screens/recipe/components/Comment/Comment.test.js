// Components
import Comment from '.'

describe('Comment', () => {
  const props = {
    avatarUri: 'https://vcdn-ngoisao.vnecdn.net/2018/01/19/1-8515-1516314392.jpg',
    name: 'Andy',
  }

  const component = mount(<Comment {...props} />)
  it('Renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  it('Renders correctly component with isGetTime = true', () => {
    component.setProps({
      isGetTime: true,
      type: 'primary',
    })

    expect(component).toMatchSnapshot()
  })

  it('Renders correctly component with type secondary', () => {
    component.setProps({
      type: 'secondary',
    })

    expect(component).toMatchSnapshot()
  })
})
