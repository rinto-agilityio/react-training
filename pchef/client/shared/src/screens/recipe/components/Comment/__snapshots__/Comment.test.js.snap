// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment renders correctly 1`] = `ReactWrapper {}`;

exports[`Comment renders correctly component with isGetTime = true 1`] = `ReactWrapper {}`;

exports[`Comment renders correctly component with type secondary 1`] = `ReactWrapper {}`;
